data_model/src/block.rs

```diff
@@ -325,10 +313,88 @@ mod candidate {
         payload: BlockPayload,
     }
 
+    #[derive(Decode, Deserialize)]
+    struct BlockPayloadCandidate {
+        header: BlockHeader,
+        transactions: Vec<CommittedTransaction>,
+        event_recommendations: Vec<EventBox>,
+    }
+
+    #[derive(Decode, Deserialize)]
+    struct BlockHeaderCandidate {
+        height: NonZeroU64,
+        prev_block_hash: Option<HashOf<SignedBlock>>,
+        transactions_hash: HashOf<MerkleTree<SignedTransaction>>,
+        creation_time_ms: u64,
+        view_change_index: u32,
+        consensus_estimation_ms: u64,
+    }
+
+    impl BlockHeaderCandidate {
+        #[cfg(feature = "std")]
+        const fn creation_time(&self) -> Duration {
+            Duration::from_millis(self.creation_time_ms)
+        }
+
+        fn validate(self) -> Result<BlockHeader, &'static str> {
+            #[cfg(feature = "std")]
+            {
+                use std::time::SystemTime;
+
+                let now = SystemTime::now()
+                    .duration_since(SystemTime::UNIX_EPOCH)
+                    .expect("INTERNAL BUG: Failed to get the current system time");
+
+                if now <= self.creation_time() {
```

Comment:
I'm not sure if this is too tight? For transactions we use `future_threshold`. @Erigara @s8sato

Reply:
What if we introduce a buffer to allow for system time differences, which is negligible relative to the average block period?

Reply:
Implying the more frequently a block is committed, the more buffer is reduced and the more likely a peer is considered faulty

Reply:
Do you mean like `future_threshold` in the queue

Reply:
Yes, it would be an analogy of `future_threshold`. Do you think there should be a common variable for the block and transaction timestamps to allow for system time differences?